lib/vanagon/platform/windows.rb

```diff
@@ -85,7 +85,37 @@ def generate_nuget_package(project)
       # @param project [Vanagon::Project] project to name
       # @return [String] name of the windows package for this project
       def nuget_package_name(project)
-        "#{project.name}-#{@architecture}-#{project.version}.#{project.release}.nupkg"
+        "#{project.name}-#{@architecture}-#{nuget_package_version(project.version, project.release)}.nupkg"
+      end
+
+      # Nuget versioning is awesome!
+      #
+      # We can only have at most four elements in the version, all must be numeric
+      # and separated by periods. We can have a prerelease element at the end of the
+      # version string, but it is distinguished by a dash and from what I can tell,
+      # must begin with a letter. We can't handle the version derived from the git
+      # version, which usually contains the short sha of the project since this short
+      # sha often contains a random selection of numbers and letters.
+      #
+      # If we have four or more elements in our version string, we assume we have commits beyond
+      # the latest tag. In that case, we assume we have a version like "1.3.2.123.2g1928uf".
+      # We just take the first four elements and hope they are numeric. Otherwise, we assume
+      # we are working from a tag, which would only contain numeric elements. We can then
+      # append the release number to the version string.
+      #
+      # @param project [Vanagon::Project] project to version
+      # @return [String] the version of the nuget package for this project
+      def nuget_package_version(version, release)
```

Comment:
nope. I couldn't get that to work. Pulling in some people who are better at ruby then I am only lead to the decision that I was trying to do things that were too complicated.